cai_causal_graph/causal_graph.py

```diff
@@ -534,6 +537,81 @@ def is_empty(self) -> bool:
         """Return True if there are no nodes and edges. False otherwise."""
         return len(self.nodes) == 0 and len(self.edges) == 0
 
+    def _check_node_exists(self, identifier: NodeLike) -> str:
+        """Check the node identifier."""
```

Comment:
This makes it sound like it just "checks" it, but in reality it also returns the checked identitified. I would change to `"""Validate and return the node identifier."""`